pkg/backup/alter_backup_schedule.go

```diff
@@ -433,9 +438,7 @@ func processFullBackupRecurrence(
 		}
 		// Copy the cadence from the incremental to the full, and delete the
 		// incremental.
-		if err := s.fullJob.SetSchedule(s.incJob.ScheduleExpr()); err != nil {
-			return scheduleDetails{}, err
-		}
+		s.fullJob.SetScheduleExpr(s.incJob.ScheduleExpr())
```

Comment:
i'm having a really hard time understanding what `processFullBackupRecurrence` does. I would have expected it to change the `FULL BACKUP crontab`, but it has turned into a dirty kitchen sink. In a follow up pr, could you clean this logic up so mortals can understand what's going on? This clean up work is less important than backup compaction stuff.

Reply:
Yeah for sure, it's definitely got a lot going on